erroneous error on goa gen

## Expected Behavior
expected to generate successfully

## Actual Behavior
goa gen github.com/dgaedcke/nmg_admin_service/design
exit status 1
service "News" method "create": field newsEntityType - defines a view but is not a result type
service "News" method "create": field postedDtTm - defines a view but is not a result type
attribute: field newsEntityType - defines a view but is not a result type
attribute: field postedDtTm - defines a view but is not a result type
make: *** [gen] Error 1

## Steps to Reproduce the Problem (including complete and simple design if relevant)
goa gen on the attached design doc

## Specifications

  - goa Version: V2
  - Platform (Linux/Windows etc.):  MacOS
[GoaV2DesignExample.zip](https://github.com/goadesign/goa/files/1764789/GoaV2DesignExample.zip)

--- a/design/attribute.go
+++ b/design/attribute.go
@@ -179,7 +179,7 @@ func (a *AttributeExpr) Validate(ctx string, parent eval.Expression) *eval.Valid
 	if views, ok := a.Metadata["view"]; ok {
 		rt, ok := a.Type.(*ResultTypeExpr)
 		if !ok {
-			verr.Add(parent, "%sdefines a view but is not a result type", ctx)
+			verr.Add(parent, "%sdefines a view %v but is not a result type", ctx, views)
 		}
 		if rt != nil {
 			found := false

--- a/dsl/result_type.go
+++ b/dsl/result_type.go
@@ -440,7 +440,9 @@ func buildView(name string, mt *design.ResultTypeExpr, at *design.AttributeExpr)
 			if dup.Metadata == nil {
 				dup.Metadata = make(map[string][]string)
 			}
-			dup.Metadata["view"] = cat.Metadata["view"]
+			if len(cat.Metadata["view"]) > 0 {
+				dup.Metadata["view"] = cat.Metadata["view"]
+			}
 			o.Set(n, dup)
 		} else if n != "links" {
 			return nil, fmt.Errorf("unknown attribute %#v", n)

